ivy/functional/frontends/paddle/tensor/linalg.py

```diff
@@ -151,3 +151,17 @@ def matrix_power(x, n, name=None):
 @to_ivy_arrays_and_back
 def cond(x, p=None, name=None):
     return ivy.cond(x, p=p, out=name)
+
+
+# dot
+@with_supported_dtypes({"2.5.0 and below": ("float32", "float64")}, "paddle")
+@to_ivy_arrays_and_back
+def dot(x, y, name=None):
+    x, y = promote_types_of_paddle_inputs(x, y)
+    out = []
+    if ivy.get_num_dims(x) == 2:
+        for xi, yi in zip(x, y):
+            out.append(ivy.vecdot(xi, yi))
+        return ivy.array(out)
```

Comment:
This code fragment doesn't look efficient. Can't you figure out a vectorized implementation? I'm pretty sure there is.

Reply:
Good point! right on it!